src/main/java/org/datacleaner/extension/elasticsearch/ElasticSearchFullSearchTransformer.java

```diff
@@ -136,22 +156,91 @@ public OutputColumns getOutputColumns() {
 
             final SearchRequestBuilder searchRequestBuilder = new SearchRequestBuilder(client)
                     .setIndices(elasticsearchDatastore.getIndexName()).setTypes(documentType).setQuery(query)
-                    .setSize(1).setSearchType(SearchType.QUERY_AND_FETCH).setExplain(true);
+                    .setSize(getNumberOfResults).setSearchType(SearchType.QUERY_AND_FETCH).setExplain(true);
 
             final SearchResponse searchResponse = searchRequestBuilder.execute().actionGet();
             final SearchHits hits = searchResponse.getHits();
-            if (hits.getTotalHits() == 0) {
+            int totalHits = (int) hits.getTotalHits();
+            if (totalHits == 0) {
+                if (allRowsCollector != null) {
+                    allRowsCollector.putValues( ArrayUtils.add(result, input));
+                }
                 return result;
+            } else {
+                //maintain compatibility of old jobs
+                if (getNumberOfResults == 1) {
+                    getFirstResult(result, input, hits);
+                    return result;
+                }
```

Comment:
Making a size of 1 a special case seems a bit messy. Wouldn't it be better to do something like this:
```java
private Object[] getResults(final hits) {
    final fetchSize = Math.min(hits, getNumberOfResults);
    final Object[] firstResult = new Object[2];
    for (int i = 0; i < fetchSize; i++) {
        if (i == 0) {
            firstResult[0] = hit.getId();
            firstResult[1] = hit.sourceAsMap();
        }
        final SearchHit hit = hits.getAt(i);
        if (allRowsCollector != null) {
            allRowsCollector.putValues(hit.getId(), hit.sourceAsMap(), input);
        }
    }
    return firstResult;
}
```

... and then simply just call and return this method from `transform`?

This would also remove a lot of code.